pfuzz/gem5_statistics/tick_statistics.py

```diff
@@ -18,20 +18,118 @@ class _Tick:
         how many instructions were executed during this tick and up to this tick
         """
 
-        tick_number: int
-        fetched_during_this_tick: int
-        decoded_during_this_tick: int
-        retired_during_this_tick: int
-        fetched_overall_by_this_tick: int
-        decoded_overall_by_this_tick: int
-        retired_overall_by_this_tick: int
+        tick_number: int = 0
+        fetched_during_this_tick: int = 0
+        decoded_during_this_tick: int = 0
+        renamed_during_this_tick: int = 0
+        dispatched_during_this_tick: int = 0
+        issued_during_this_tick: int = 0
+        completed_during_this_tick: int = 0
+        retired_during_this_tick: int = 0
+        stored_during_this_tick: int = 0
 
-    def get_tick_statistics_list(self, path_to_trace: str) -> List[_Tick]:
+        fetched_so_far: int = 0
+        decoded_so_far: int = 0
+        renamed_so_far: int = 0
+        dispatched_so_far: int = 0
+        issued_so_far: int = 0
+        completed_so_far: int = 0
+        retired_so_far: int = 0
+        stored_so_far: int = 0
+
+    def update_so_far_stages_numbers(
+        self, ticks: Dict[str, _Tick], tick_number: str
+    ) -> None:
+        ticks[tick_number].fetched_so_far = ticks["so_far"].fetched_so_far
+        ticks[tick_number].decoded_so_far = ticks["so_far"].decoded_so_far
+        ticks[tick_number].renamed_so_far = ticks["so_far"].renamed_so_far
+        ticks[tick_number].dispatched_so_far = ticks["so_far"].dispatched_so_far
+        ticks[tick_number].issued_so_far = ticks["so_far"].issued_so_far
+        ticks[tick_number].completed_so_far = ticks["so_far"].completed_so_far
+        ticks[tick_number].retired_so_far = ticks["so_far"].retired_so_far
+        ticks[tick_number].stored_so_far = ticks["so_far"].stored_so_far
+
+    def proccess_stage(
+        self, stage: str, ticks: Dict[str, _Tick], current_tick: str
+    ) -> str:
+        stage = stage.removeprefix("O3PipeView:")
+        prefix = stage[0:3]
+        tick_number = (
+            stage.split(":")[1].rstrip()
+            if stage.split(":")[1].rstrip() != "0"
+            else current_tick
+        )
+
+        if tick_number not in ticks:
+            ticks[tick_number] = self._Tick()
+
+        if prefix == "fet":
```

Comment:
```suggestion
        if stage.startswith("fet"):
```
Писать, наверное, подольше, но тогда выше срез не нужен.

Reply:
Со срезом более аккуратно и читаемо имхо